prow/tide/tide.go

```diff
@@ -569,34 +576,37 @@ func toSimpleState(s kube.ProwJobState) simpleState {
 
 // isPassingTests returns whether or not all contexts set on the PR except for
 // the tide pool context are passing.
-func isPassingTests(log *logrus.Entry, ghc githubClient, pr PullRequest) bool {
+func isPassingTests(log *logrus.Entry, ghc githubClient, pr PullRequest, cc contextChecker) bool {
 	log = log.WithFields(pr.logFields())
 	contexts, err := headContexts(log, ghc, &pr)
 	if err != nil {
 		log.WithError(err).Error("Getting head commit status contexts.")
 		// If we can't get the status of the commit, assume that it is failing.
 		return false
 	}
-	return len(unsuccessfulContexts(contexts)) == 0
+	return len(unsuccessfulContexts(contexts, cc)) == 0
 }
 
-// unsuccessfulContexts determines which contexts from the list are failed that
-// we care about. For instance, we do not care about our own context.
-func unsuccessfulContexts(contexts []Context) []Context {
+// unsuccessfulContexts determines which contexts from the list that we care about are
+// failed. For instance, we do not care about our own context.
+// If the branchProtection is set to only check for required checks, we will skip
+// all non-required tests. If required tests are missing from the list, they will be
+// added to the list of failed contexts.
+func unsuccessfulContexts(contexts []Context, cc contextChecker) []Context {
 	var failed []Context
 	for _, ctx := range contexts {
-		if string(ctx.Context) == statusContext {
+		if cc.ignoreContext(ctx) {
```

Comment:
this seems like a regression if we mark the `tide` status required

Reply:
cc already ignores tide